### PR TITLE
Adds tox and test requirements file

### PR DESCRIPTION
**Pull request by benjwadams**
Adds tox so multiple versions of Python can be tested on local machines.
Does not work with `conda`.  Also added test_requirements.txt file
for testing dependencies.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[tox]
+envlist = py27,py33
```
Comment (daf) [inline tox.ini]: should align this with travis-ci, which uses 3.4 and 3.5, no 3.3.  i looked briefly at tox docs and didn't see a `py35` entry?

Comment (benjwadams) [inline tox.ini]: Will check this out and modify as needed.

Comment (benjwadams) [inline tox.ini]: So it looks possible.  However, supporting multiple minor versions of Python with the same major version (i.e. 3.4 and 3.5) might prove to be a little more complicated and have some OS specific steps, assuming one hasn't compiled from source to a directory such as `/opt/`

Comment (daf) [inline tox.ini]: ok, err to the future (3.5) if that makes this easier.  what are the issues (summary level)?

Comment (daf) [inline tox.ini]: poke

**Comment by lukecampbell** [comment]
@daf were you able to run tox? I saw a ton of issues last I tried.

**Comment by daf** [comment]
I modified the `tox.ini` locally to do `py27`, `py34`, and `py35`.  `py34` required I installed ubuntu package `libpython3.4-dev` before it worked.  My current ubuntu release does not have an equivalent 3.5 variety, so that part is broken for now, but the new ubuntu just dropped and I think that has it.  Not sure what that means for Travis, though.

**Comment by lukecampbell** [comment]
closed in favor of https://github.com/ioos/compliance-checker/issues/279

**Pull request closed without merging**